packages/grid/_modules_/grid/components/cell/GridEditSelectCell.tsx

```diff
@@ -0,0 +1,77 @@
+import * as React from 'react';
+import Select, { SelectProps } from '@material-ui/core/Select';
+import MenuItem from '@material-ui/core/MenuItem';
+import { GridCellParams } from '../../models/params/gridCellParams';
+import { GRID_CELL_EDIT_PROPS_CHANGE_COMMITTED } from '../../constants/eventsConstants';
+
+export function GridEditSelectCell(props: GridCellParams & SelectProps) {
+  const {
+    id,
+    value,
+    formattedValue,
+    api,
+    field,
+    row,
+    colDef,
+    cellMode,
+    isEditable,
+    className,
+    getValue,
+    hasFocus,
+    ...other
+  } = props;
+
+  const [valueState, setValueState] = React.useState(value);
+
+  const handleChange = React.useCallback(
+    (event) => {
+      const newValue = event.target.value;
+      const editProps = { value: newValue };
+
+      setValueState(newValue);
+      api.publishEvent(
+        GRID_CELL_EDIT_PROPS_CHANGE_COMMITTED,
+        { id, field, props: editProps },
+        event,
+      );
+      api.setCellMode(id, field, 'view');
```

Comment:
Selecting an item with the keyboard is not closing the select. I had to use a different approach to save the value in [`renderEditProgress`](https://github.com/mui-org/material-ui-x/blob/master/packages/grid/x-grid-data-generator/src/renderer/renderEditProgress.tsx). In #1955 I improved this API to make very clear what each methods does.

Reply:
Yes, I saw the PR. maybe it makes sense to wait with this one until the other one is merged because I would have to change the way the cell edit mode is handled.

Reply:
Ops, I linked the wrong component. [`renderEditStatus`](https://github.com/mui-org/material-ui-x/blob/master/packages/grid/x-grid-data-generator/src/renderer/renderEditStatus.tsx) is based on Select.

Reply:
Fixed